walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +274,178 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  // Mandatory and optional attendee tests.
+  @Test
+  public void onlyMandatoryAttendeesAreConsidered() {
+    // Have each person have different events. We should see three options because the mandatory 
+    // attendee's (A & B) have non-overlapping events while the optional attendee (C) is not 
+    // available at any point in the day.
+    //
+    // Events  :       |--A--|     |--B--|
+    //           |--------------C--------------|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, 
+                                                    TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
```

Comment:
Something Hiram noted in my PR is that when writing tests, including parameter names as a small comment where they may be ambiguous is helpful. For example here, it's unclear what the booleans in `fromStartEnd` could represent, so you can indicate them as,

`TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, /*inclusive=*/false)`

Reply:
Ahhh yes thank you for pointing this out! I seen/done this before in C++ and Python but did not even think of extending that to Java this summer.